query-algebrizer/src/clauses/not.rs

```diff
@@ -0,0 +1,63 @@
+// // Copyright 2016 Mozilla
+// //
+// // Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+// // this file except in compliance with the License. You may obtain a copy of the
+// // License at http://www.apache.org/licenses/LICENSE-2.0
+// // Unless required by applicable law or agreed to in writing, software distributed
+// // under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+// // CONDITIONS OF ANY KIND, either express or implied. See the License for the
+// // specific language governing permissions and limitations under the License.
+use mentat_core::{
+    Schema,
+};
+
+use mentat_query::{
+    NotJoin,
+    UnifyVars,
+    ContainsVariables,
+};
+
+use clauses::ConjoiningClauses;
+
+use errors::{
+    Result,
+};
+
+use types::{
+    ColumnConstraint,
+    ComputedTable,
+    PushComputed,
+    SourceAlias,
+};
+
+impl ConjoiningClauses {
+    pub fn apply_not_join(&mut self, schema: &Schema, not_join: NotJoin) -> Result<()> {
+        let projected = match not_join.unify_vars {
+            UnifyVars::Implicit => not_join.collect_mentioned_variables ().into_iter().collect(),
+            UnifyVars::Explicit(vs) => vs.into_iter().collect(),
+        };
+        let template = self.use_as_template(&projected);
+
+        let mut receptacle = template.make_receptacle();
```

Comment:
If you're only doing one, you can use the `template` directly. You'll note that `use_as_template` is essentially an intersected `clone`, and `make_receptacle` is a full `clone` of the same parts.